src/sv-pipeline/scripts/make_evidence_qc_table.py

```diff
@@ -0,0 +1,264 @@
+    Reads MELT insert size into an array.
+    Args:
+        filename: melt_insert_size in sample data table
+
+    Returns:
+        Array[Float] of melt_insert_size for each sample
+    """
+    insert_size = pd.read_csv(filename, sep="\t")
+    melt_insert_size_array = pd.DataFrame(insert_size, columns=["melt_insert_size"]).astype(float)
+    return melt_insert_size_array
+
+
+def get_col_name(caller: str, outlier_type: str) -> str:
+    return f"{caller}_{outlier_type}_outlier"
+
+
+def read_outlier(filename: str, outlier_col_label: str, idx=10) -> pd.DataFrame:
+    """
+
+    Args:
+        filename: A tab-delimited file containing the output of
+        Evidence QC on a caller (e.g., Manta) with low/high.
+
+        outlier_col_label: Column header for the outliers.
+
+        idx: The index of the cell in the array containing sample
+        URI split by `/` that contains sample ID.
+    Returns:
+        A pandas DataFrame containing the number of times a sample
+        appears in the QC of a caller output.
+    """
+    df = pd.read_csv(filename, sep="\t")
+    if df["Outlier_Sample"].empty:
+        df[ID_COL] = df.apply(lambda _: "NA", axis=0)
+        outlier_sample = df.pivot_table(columns=[ID_COL], aggfunc="size").astype(int)
+    else:
+        df[ID_COL] = df["Outlier_Sample"].str.split("/", expand=True)[idx].str.split(".", expand=True)[0]
+        outlier_sample = df.pivot_table(columns=[ID_COL], aggfunc="size").astype(int)
+    outlier_df = outlier_sample.reset_index()
+    outlier_df.columns = [ID_COL, outlier_col_label]
+    return outlier_df
+
+
+def read_all_outlier(filename_manta: str, filename_melt: str, filename_wham: str, outlier_type: str) -> pd.DataFrame:
+    """
+
+    Args:
+        filename_manta: Outliers determined in EvidenceQC for Manta.
+        filename_melt: Outliers determined in EvidenceQC for MELT.
+        filename_wham: Outliers determined in EvidenceQC for Wham.
+        outlier_type:
+
+    Returns:
+        The total number of times that a sample appears as an outlier
+        across all four algorithms for each sample.
+    """
+    # Manta:
+    col_name = get_col_name("manta", outlier_type)
+    outlier_manta_df = read_outlier(filename_manta, col_name)
+    dict_manta = dict(list(zip(outlier_manta_df[ID_COL], outlier_manta_df[col_name])))
+
+    # Melt:
+    col_name = get_col_name("melt", outlier_type)
+    outlier_melt_df = read_outlier(filename_melt, col_name)
+    dict_melt = dict(list(zip(outlier_melt_df[ID_COL], outlier_melt_df[col_name])))
+
+    # Wham:
+    col_name = get_col_name("wham", outlier_type)
+    outlier_wham_df = read_outlier(filename_wham, col_name, 11)
+    dict_wham = dict(list(zip(outlier_wham_df[ID_COL], outlier_wham_df[col_name])))
+
+    # merging all the dictionaries
+    outlier_dicts = [dict_manta, dict_melt, dict_wham]
+    merged_dicts = Counter()
+    for counted in outlier_dicts:
+        merged_dicts.update(counted)
+    all_outliers = dict(merged_dicts)
```

Comment:
When I did this I called `Counter()` on the concatenated sample ID columns from all the high outliers files, which seemed a little simpler/marginally more efficient because there were no dictionaries and you could count all of the callers at once